### DISCUSSION TITLE
Error add garbage record to the "oo_plugin_form_fieldconfig" table when saving all forms.

### DISCUSSION THREAD

**Discussion by dai-eastgate** [onOffice-Web-Org/oo-wp-plugin#461] [_Discarded pitches]
@fredericalpers When I fixed and tested the bug https://github.com/onOffice-Web-Org/oo-wp-plugin/discussions/454
1. I found an error:
- When "add new" or "update" action of All Form.
Each time we click "save changes" in the form, we will add a record for "form_Id = 0" to the "oo_plugin_form_fieldconfig" table. Meanwhile, there is no existing form with "id=0".

![bug1](https://user-images.githubusercontent.com/106214469/224595479-63b13245-ddd5-4161-81c2-b88f1b381f13.png)

2. The cause of the error:
This code always has the data for "availableOptions" that will add to the database when we save the form.

![bug-02](https://user-images.githubusercontent.com/106214469/224595824-e3c2ec0e-fd8e-4f24-b868-a839090dc524.png)

3. But I don't understand What is the function "Reduce values according to selected filter"(availableOptions) of the checkbox in the field?
Currently, when I click the "Reduce values ​​according to selected filter" checkbox and click "Save Changes" it is not saved in the database. And in the backend, the checkbox "Reduce values ​​according to selected filter" is also unchecked

![image](https://user-images.githubusercontent.com/106214469/224640253-dc7cd32a-7720-4149-a961-22c6aea3009d.png)

Can you explain it to me?
So that I can find a solution to solve this issue.

**Comment by fredericalpers** [comment]
@dai-eastgate I'm not sure why the field "Reduce values according to selected filter" is available in forms. Usually it's supposed to reduce the values from all possible selections (kind of property) to only the selections made available by the customers onOffice enterprise. E.g. it might be reduced to just Penthouses and Villas instead of all different kind of apartments and houses.

I'll discuss this with one of our developers and update you afterwards.

**Reply by fredericalpers** [reply]
@dai-eastgate The option "reduce values according to selected filter" should not be available in forms. I found the same bug before as well. I will write a new pitch.

**Reply by dai-eastgate** [reply]
same error in https://github.com/onOffice-Web-Org/oo-wp-plugin/discussions/464
